docs/best-practices/elastic-deployment.md

```diff
@@ -0,0 +1,457 @@
+This CRD can give existing/incremental Workloads the ability of multi domain dispersion, elastic schedule and fine management in a non intrusive manner.
+
+In this page, we will take a simple web application as an example to help users build an automatic extreme elastic scheduling solution, combining with WorkloadSpread, KEDA, Prometheus and Alibaba Cloud Elastic Instances (ECI).
+
+## Introduction 
+
+### Architecture
+
+The architecture of this solution is as follows:
+![arch](../../static/img/docs/user-manuals/elasticd-deployment-arch.jpg)
+
+**Special Note:**
+- In the solution, HPA is managed through KEDA. KEDA is an enhanced autoscaling component based on HPA. Compared with the native HPA, KEDA has much richer user-defined metrics.
+  
+- We take a trick that we collects the metrics of Nginx instead of Web Pod, because we want to reuse the open-source Nginx-Prometheus-Exporter. It's easier to use this exporter to explore the number of https links and other metrics. Most importantly, the traffic entering the Web Pod must go through the Niginx Ingress. Therefore, we are going to directly use the metrics of Nginx to combine KEDA to implement automatic scale up and down.
+
+- At least version 1.21 is required by WorkloadSpread to manage Deployment, but ACK Kubernetes clusters currently supports up to version 1.20. Therefore, we takes CloneSet as an example in the solution.
+
+
+### Goals
+Our goal is to fully automate the following behaviors:
+
+- If the traffic exceeded the threshold within a certain time window (the ** traffic ** here is defined as the number of http connections per second, which can be defined according to actual needs), it will scale up replicas automatically;
+
+- When scaling up, the higher priority will be given to the fixed resource pool to schedule pod. When the fixed resource pool is insufficient or reached the `MaxReplicas` limit, the Pods will be automatically scheduled to the elastic resource pool;
+
+- When the traffic is lower than the threshold within a certain time window,  it will scale down replicas automatically;
+
+- When scaling down, the Pods in the elastic resource pool will be deleted first.
+
+## Dependency Installation 
+We use a ACK Kubernetes Cluster with 3 ECS nodes and 1 Virtual-Kubelet (VK) nodes. ECS nodes corresponds to the fixed resource pool, and VK node corresponds to the elastic resource pool.
+```shell
+$ k get node
+NAME                         STATUS   ROLES    AGE    VERSION
+us-west-1.192.168.0.47       Ready    <none>   153d   v1.20.11-aliyun.1
+us-west-1.192.168.0.48       Ready    <none>   153d   v1.20.11-aliyun.1
+us-west-1.192.168.0.49       Ready    <none>   153d   v1.20.11-aliyun.1
+virtual-kubelet-us-west-1a   Ready    agent    19d    v1.20.11-aliyun.1 
+```
+
+### Installing OpenKruise
+```shell
+$ helm repo add openkruise https://openkruise.github.io/charts/
+
+$ helm repo update
+
+$ helm install kruise openkruise/kruise --version 1.0.0
```

Comment:
You'd better not to paste the version to install, otherwise we have to modify it every time Kruise has a new version.
Just a link of installation doc is enough.